### PR TITLE
Add lld16 package to Mariner

### PR DESCRIPTION
**Pull request by sameluch**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add lld16 package to Mariner alongside the llvm16 and clang16

###### Change Log  

- Add lld16 package to Mariner alongside the llvm16 and clang16

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local builds
- Buddy Build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=404509&view=results

### CODE REVIEW

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 1]
```diff
@@ -2207,6 +2207,7 @@
                 "livepatch-5.15.98.1-1.cm2",
                 "livepatching",
                 "lld",
+				"lld16",
```
Comment (mfrw) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:2210]: NIT: formating looks a bit off here.

Comment (sameluch) [reply SPECS/LICENSES-AND-NOTICES/data/licenses.json]: Fixed

File: SPECS/lld/lld16.spec [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+%build
+mkdir -p build
+cd build
+%cmake ..                                                         \
+       -G Ninja                                                   \
+       -DCMAKE_BUILD_TYPE=Release                                 \
+       -DCMAKE_SKIP_RPATH:BOOL=on                                 \
+       -DCMAKE_C_FLAGS=-I../../libunwind-%{version}.src/include   \
+       -DCMAKE_CXX_FLAGS=-I../../libunwind-%{version}.src/include \
+	   -DLLVM_LINK_LLVM_DYLIB:BOOL=on                             \
```
Comment (jslobodzian) [inline SPECS/lld/lld16.spec:61]: nit.  extra tab or spacing here.

**Review by jslobodzian — APPROVED** [review]

File: SPECS/lld/lld16.spec (hunk 2 above)
Comment (sameluch) [reply SPECS/lld/lld16.spec]: fixed spacing for this line

**Pull request merged**